### ISSUE TITLE
Need to bump engine.io-client once memory leak issue is fixed

### ISSUE THREAD

**Issue by AlmirKadric** [primus/primus#495]
**Version:** https://github.com/primus/primus/commit/0458cfdeaae0e80df310bb97d0a63e8fddd52c3f

**Environment:**
  - **Operating system:** MacOSX
  - **Browser:** Chrome
  - **Node.js:** 5.11.1

**Expected result:** Engine.IO client should be at version 1.6.9 according to this commit: https://github.com/primus/primus/commit/1126671ea0d8efd76b51cd44834570806930bf1d

**Actual result:**
Engine.IO client is actually quite old pre 1.6.0 and is missing vital fixes, including this one which causes memory leaks: https://github.com/socketio/engine.io/commit/deb7ae40a4e19016d0b4be52b57d77db89cbd846#diff-9b2f90c89d460ee80ced18e01748824eR198

**Steps to reproduce:** Irrelavent

**EDIT:** Seems this is an issue of engine.io-client repo not being in sync with engine.io (server) repo. Waiting on https://github.com/socketio/engine.io-client/issues/493 being resolved.

**Comment by lpinca** [comment]
Your link points to the server-side engine.io.
I've also run the script that we use to update the client-side lib and it seems that there no changes.

```shell
$ ./transformers/engine.io/update.sh 
```

We are up to date.

**Comment by lpinca** [comment]
If you wonder why there is `engine.io-client@1.6.10` and `engine.io-client@1.6.11`, it is because they keep the client and server versions in [sync](https://github.com/socketio/engine.io/blob/2c0ae2972b20f6e843dffea17d11262adee3a381/test/engine.io.js#L23-L27).

**Comment by AlmirKadric** [comment]
Wow, you're absolutely right.
Created an issue on their client repo here: https://github.com/socketio/engine.io-client/issues/493

Hopefully they resolve this and we can upgrade the client to include all missing fixes.
Mind if we leave this open till that is resolved?

**Comment by lpinca** [comment]
I have no problem keeping this open but again the fix you are talking about is in `engine.io` not `engine.io-client` and `engine.io@1.6.11` already includes the [fix](https://github.com/socketio/engine.io/blob/1.6.11/lib/socket.js#L224-L259) linked.

**Comment by AlmirKadric** [comment]
Exactly, the issue is that engine.io-client doesn't have this fix when it should, which in turn means that Primus does not have this fix in their locally cached copy of the client. Which is why I have linked the issue on socket.io-client here so that once that is resolved, this ticket can be resolved as well. Will update the title to be more descriptive.

**Comment by lpinca** [comment]
Are you saying that the same fix should be ported to `engine.io-client` as well? I'm not sure I follow.

**Comment by AlmirKadric** [comment]
Sorry if I wasn't clear @lpinca.
Yes that is correct, the `engine.io-client` should also contain that fix. If not there will be registered event handlers not being cleaned up after a failed upgrade attempt which means that the socket object will not be garbage collected.

Originally when I made the issue I did not realise that the `engine.io-client` repo did not contain the fix which is what `primus` is using. I apologise for that misunderstanding. Also I am currently writing this under the assumption that the socket.js file on both server and client repo's should be in sync.

Edit: after looking at this again it seems that though some logic is shared, there are quite a lot of differences between the server and client code. Also looking at my issue, seems that it may in fact be already resolved with this line https://github.com/socketio/engine.io-client/blob/master/lib/socket.js#L312

I guess this begs the question is the socket.js file supposed to be in sync between the server and client repos, but I guess this issue is resolved.
